src/Admin/AbstractAdminExtension.php

```diff
@@ -24,6 +24,9 @@
 
 /**
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * @phpstan-template T of object
+ * @template-implements AdminExtensionInterface<T>
```

Comment:
👎 

An admin extension can be used for different classes that does not explicitly require the same interface.

Reply:
Cant the user write AdminExtensionInterface<object> or AdminExtensionInterface<Foo|Bar> in these cases

Reply:
Is this valid for PHP generics?

Reply:
I never used this in this special case before. But I have trouble to see why it wouldn't be

`array<int|string>` is valid for instance.

Reply:
If it is not generic, I think at least we should remove `AdminInterface<object>` references because apparently once defined, you cannot substitute `object` with another type (I'm still struggling with generics, so maybe I'm doing something wrong): 

https://phpstan.org/r/7beb7747-14f8-4dac-8f87-690cfb93c65b

Reply:
I would recommend to remove `<object>` and maybe report the issue to phpstan in order to know what to do.

Reply:
Another way I guess would be to add generics and in those classes that use different interfaces, they could be defined like this:
```php
/**
 * @template-implements AdminExtensionInterface<object>
 */
```
or
```php
/**
 * @template-extends AbstractAdminExtension<object>
 */
```
And it would be the same result as it is right now.

Reply:
You're right. This allows using specific types or use `object` as a fallback, unless this is supported by psalm/phpstan